File: src/brs/http/GetAssetIds.java
package brs.http;

import brs.Asset;
import brs.assetexchange.AssetExchange;
import brs.util.CollectionWithIndex;
import brs.util.Convert;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import javax.servlet.http.HttpServletRequest;

import static brs.http.common.Parameters.FIRST_INDEX_PARAMETER;
import static brs.http.common.Parameters.LAST_INDEX_PARAMETER;
import static brs.http.common.ResultFields.ASSET_IDS_RESPONSE;
import static brs.http.common.ResultFields.NEXT_INDEX_RESPONSE;

public final class GetAssetIds extends APIServlet.JsonRequestHandler {

  private final AssetExchange assetExchange;

  public GetAssetIds(AssetExchange assetExchange) {
    super(new APITag[] {APITag.AE}, FIRST_INDEX_PARAMETER, LAST_INDEX_PARAMETER);
    this.assetExchange = assetExchange;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) {

    int firstIndex = ParameterParser.getFirstIndex(req);
    int lastIndex = ParameterParser.getLastIndex(req);

    JsonArray assetIds = new JsonArray();
    CollectionWithIndex<Asset> assets = assetExchange.getAllAssets(firstIndex, lastIndex);
    for (Asset asset : assets) {
      assetIds.add(Convert.toUnsignedLong(asset.getId()));
    }
    JsonObject response = new JsonObject();
    response.add(ASSET_IDS_RESPONSE, assetIds);
    
    if(assets.hasNextIndex()) {
      response.addProperty(NEXT_INDEX_RESPONSE, assets.nextIndex());
    }
    
    return response;
  }

}

File: src/main/java/brs/http/GetAssetIds.kt
package brs.http

import brs.assetexchange.AssetExchange
import brs.http.common.Parameters.FIRST_INDEX_PARAMETER
import brs.http.common.Parameters.LAST_INDEX_PARAMETER
import brs.http.common.ResultFields.ASSET_IDS_RESPONSE
import brs.util.toUnsignedString
import com.google.gson.JsonArray
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import javax.servlet.http.HttpServletRequest

internal class GetAssetIds(private val assetExchange: AssetExchange) : APIServlet.JsonRequestHandler(arrayOf(APITag.AE), FIRST_INDEX_PARAMETER, LAST_INDEX_PARAMETER) {

    override suspend fun processRequest(request: HttpServletRequest): JsonElement {

        val firstIndex = ParameterParser.getFirstIndex(request)
        val lastIndex = ParameterParser.getLastIndex(request)

        val assetIds = JsonArray()
        for (asset in assetExchange.getAllAssets(firstIndex, lastIndex)) {
            assetIds.add(asset.id.toUnsignedString())
        }
        val response = JsonObject()
        response.add(ASSET_IDS_RESPONSE, assetIds)
        return response
    }

}
